src/utils.cpp

```diff
@@ -37,36 +38,129 @@ std::vector<int> read_cpu_range(std::string path)
 
 namespace bpftrace {
 
-static bool usdt_probe_cached = false;
-static std::map<std::string, usdt_probe_pair> usdt_probe_cache_outer;
+static bool provider_cache_loaded = false;
```

Comment:
The way we have this cache implemented today doesn't support attaching to probes on multiple paths, right? For example: attaching to multiple executables, like Python + PostgreSQL.

Reply:
Correct, we would need to create a cache by path for that I believe. Right now if you load by process and it has multiple libraries with probes, it will cache all of them.

There is no way to specify multiple paths / a wildcard of paths right now though.